sox/file_info.py

```diff
@@ -234,17 +236,18 @@ def silent(input_filepath, threshold=0.001):
         return True
 
 
-def validate_input_file(input_filepath):
+def validate_input_file(input_filepath: Union[str, Path]) -> None:
     '''Input file validation function. Checks that file exists and can be
     processed by SoX.
 
     Parameters
     ----------
-    input_filepath : str
+    input_filepath : path-like (str or pathlib.Path)
         The input filepath.
 
     '''
-    if not os.path.exists(input_filepath):
+    input_filepath = Path(input_filepath)
+    if not input_filepath.exists():
```

Comment:
❤️

Reply:
what about `os.access(str(input_filepath), os.R_OK)` ?

Reply:
@lostanlen sure we could check for that too! Maybe in a separate PR after this one?